oteapi_dlite/strategies/oceanlab_influx_parser.py

```diff
@@ -0,0 +1,279 @@
+"""Strategy for oceanlab data parsing from Influx DB."""
+
+import sys
+from typing import Annotated, Optional
+
+import cachetools  # type: ignore
+import dlite
+import influxdb_client
+import jinja2
+from fastapi import logger
+from oteapi.models import AttrDict, HostlessAnyUrl, ParserConfig, ResourceConfig
+from pandas import DataFrame
+from pydantic import Field, SecretStr
+from pydantic.dataclasses import dataclass
+
+from oteapi_dlite.models import DLiteSessionUpdate
+from oteapi_dlite.utils import get_collection, update_collection
+from oteapi_dlite.utils.utils import get_meta
+
+if sys.version_info >= (3, 10):
+    from typing import Literal
+else:
+    from typing_extensions import Literal
+
+
+class InfluxParseParseConfig(AttrDict):
+    """Configuration for DLite Excel parser."""
+
+    id: Annotated[
+        Optional[str], Field(description="Optional id on new instance.")
+    ] = None
+
+    label: Annotated[
+        Optional[str],
+        Field(
+            description="Optional label for new instance in collection.",
+        ),
+    ] = "json-data"
+
+    resourceType: Annotated[
+        Optional[Literal["resource/url"]],
+        Field(
+            description=ResourceConfig.model_fields["resourceType"].description,
+        ),
+    ] = "resource/url"
+    downloadUrl: Annotated[
+        Optional[HostlessAnyUrl],
+        Field(
+            description=ResourceConfig.model_fields["downloadUrl"].description,
+        ),
+    ] = None
+    mediaType: Annotated[
+        Optional[str],
+        Field(
+            description=ResourceConfig.model_fields["mediaType"].description,
+        ),
+    ] = None
+    storage_path: Annotated[
+        Optional[str],
+        Field(
+            description="Path to metadata storage",
+        ),
+    ] = None
+    collection_id: Annotated[
+        Optional[str], Field(description="A reference to a DLite collection.")
+    ] = None
+
+    url: Annotated[Optional[str], Field(description="url to the db")] = None
+
+    USER: Annotated[Optional[str], Field(description="user to the db")] = None
+
+    PASSWORD: Annotated[
+        Optional[SecretStr], Field(description="user pwd to the db")
+    ] = None
+
+    DATABASE: Annotated[Optional[str], Field(description="database name")] = (
+        None
+    )
+
+    RETPOLICY: Annotated[
+        Optional[str], Field(description="retpolicy for db")
+    ] = None
+
+    time_range: Annotated[
+        str, Field(description="timerange of values. eg : -12h")
+    ] = "-12h"
+
+    size_limit: Annotated[str, Field(description="rows to be extracted")] = "50"
```

Comment:
It makes sense to make this an integer, no?
Also, I've added `ge` (greater or equal to). I'm guessing 0 is a fine value, or should it always be larger than 0? If the latter is the case, one can either use `gt` instead or change the value here to 1.
```suggestion
    size_limit: Annotated[int, Field(description="rows to be extracted", ge=0)] = 50
```